clang/lib/Basic/Targets/X86.cpp

```diff
@@ -119,9 +119,13 @@ bool X86TargetInfo::initFeatureMap(
     setFeatureEnabled(Features, F, true);
 
   std::vector<std::string> UpdatedFeaturesVec;
-  bool HasEVEX512 = true;
+  std::vector<std::string> UpdatedAVX10FeaturesVec;
+  int HasEVEX512 = -1;
```

Comment:
Use std::optional<bool> ?

Reply:
I think it's better to use enum. It's a 3-status flag. std::optional isn't much useful here.